tests/test_visualizations.py

```diff
@@ -0,0 +1,308 @@
+        assert (actual == expected)
+
+    def test_description_with_single_quotes_and_colon(self):
+        raw = ("Layer 'auto: Filter by HU' loaded by 1.25")
+        expected = 'Layer auto: Filter by HU loaded by 1.25'
+        assert visualizations._format_description(raw) == expected
+
+
+class TestDateFieldFormatter:
+    def test_format_DateField(self):
+        expected = '2015-06-01'
+        date = datetime(2015, 6, 1, 12, 11, 10)
+        assert (visualizations._format_DateField(date) == expected)
+
+
+class TestMoneyFieldFormatter:
+    def test_small_MoneyField(self):
+        m = {'currency': 'USD', 'value': 123.11}
+        mf = convert_to_analyzere_object(m)
+        expected = '123 USD'
+        assert visualizations._format_MoneyField(mf) == expected
+
+    def test_large_MoneyField(self):
+        m = {'currency': 'USD', 'value': 123456789.11}
+        mf = convert_to_analyzere_object(m)
+        expected = '123,456,789 USD'
+        assert visualizations._format_MoneyField(mf) == expected
+
+    def test_unlimited_MoneyField(self):
+        m = {'currency': 'USD', 'value': float_info.max}
+        mf = convert_to_analyzere_object(m)
+        expected = 'unlimited'
+        assert visualizations._format_MoneyField(mf) == expected
+
+class TestLayerTermsFormatter:
+    def test_layer(self, layer_view):
+        t,w = visualizations._format_layer_terms(layer_view)
+        assert w == False
+
+    def test_attachment_warning(self, layer_view_attachment_warning):
+        t, w = visualizations._format_layer_terms(
+            layer_view_attachment_warning)
+
+    def test_filter_warning(self, layer_view_filter_warning):
+        t, w = visualizations._format_layer_terms(layer_view_filter_warning)
+        assert w == True
+
+    def test_participation_warning(self, layer_view_participation_warning):
+        t, w = visualizations._format_layer_terms(
+            layer_view_participation_warning)
+        assert w == True
+
+
+class TestLayerViewDigraph:
+    def test_invalid_lv(self):
+        m = {'something': 'here'}
+        lv = convert_to_analyzere_object(m, Resource)
+        with pytest.raises(ValueError):
+            lvg = visualizations.LayerViewDigraph(lv)
+
+    def test_basic(self, layer_view):
+        lvg = visualizations.LayerViewDigraph(layer_view)
+
+    def test_without_terms(self, layer_view):
+        lvg = visualizations.LayerViewDigraph(layer_view, with_terms=False)
+        assert lvg._with_terms == False
+
+    def test_with_verbose(self, layer_view):
+        lvg = visualizations.LayerViewDigraph(layer_view, verbose=True)
+        assert lvg._verbose == True
+
+    def test_format(self, layer_view):
+        lvg = visualizations.LayerViewDigraph(layer_view, format='pdf')
+        assert lvg._graph.format == 'pdf'
+        lvg = visualizations.LayerViewDigraph(layer_view, format='svg')
+        assert lvg._graph.format == 'svg'
+
+    def test_render(self, layer_view):
```

Comment:
Not sure how much pain this would involve, but to avoid having to write to the file system in tests and clean up, using Mock() for the _graph member and just checking the arguments to _graph.render() or whatever that the correct filename is used should be enough... maybe?

Assuming we can trust the graphviz library to do what it promises... but we should be able to.